coordinator/src/node/unrealized_pnl.rs

```diff
@@ -0,0 +1,68 @@
+use crate::db;
+use crate::node::Node;
+use crate::position::models::Position;
+use anyhow::Context;
+use anyhow::Result;
+use diesel::r2d2::ConnectionManager;
+use diesel::r2d2::PooledConnection;
+use diesel::PgConnection;
+use rust_decimal::Decimal;
+use time::OffsetDateTime;
+use trade::bitmex_client::BitmexClient;
+use trade::bitmex_client::Quote;
+use trade::cfd::calculate_pnl;
+use trade::Direction;
+
+pub async fn sync(node: Node) -> Result<()> {
+    let mut conn = node.pool.get()?;
+
+    let positions = db::positions::Position::get_all_open_or_closing_positions(&mut conn)?;
+    let current_quote = BitmexClient::get_quote(&OffsetDateTime::now_utc())
+        .await
+        .context("Failed to fetch quote from BitMEX")?;
+
+    for position in positions.iter() {
+        if let Err(e) = sync_position(&mut conn, position, current_quote.clone()) {
+            tracing::error!(position_id=%position.id, ?current_quote, "Failed to update position's unrealized pnl in database: {e:#}")
+        }
+    }
+
+    Ok(())
+}
+
+fn sync_position(
+    conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+    position: &Position,
+    quote: Quote,
+) -> Result<()> {
+    let current_price = match position.direction {
+        trade::Direction::Long => quote.bid_price,
+        trade::Direction::Short => quote.ask_price,
+    };
+
+    let average_entry_price = Decimal::try_from(position.average_entry_price)
+        .context("Failed to convert average entry price to Decimal")?;
+
+    let (long_leverage, short_leverage) = match position.direction {
+        Direction::Long => (position.leverage, 1.0_f32),
+        Direction::Short => (1.0_f32, position.leverage),
+    };
```

Comment:
It's a bit scary that this kind of logic is located in this module. It's probably hard to define where this should be located atm. Maybe we can leave a comment talking about how the position is the trader's position and the coordinator is always leverage 1.

Reply:
I did a refactoring where I moved the `coordinator_pnl` onto the `Position` struct and added tests.
This particular bit remains within that pnl calculation function. Eventually we might also want to extract this.

Instead of using `1.0` I opted for using the `COORDINATOR_LEVERAGE` constant that I made public in the coordinator's `node` module.